patroni/ha.py

```diff
@@ -128,6 +128,13 @@ def is_leader(self):
         with self._is_leader_lock:
             return self._is_leader > time.time()
 
+    def is_static_single_node(self):
+        if self.patroni.config is None:
+            return False
+        # QUESTION to reviewers: should we check the last held DCS config as well just to ensure there is no conflict?
```

Comment:
I think it should not be a local configuration, otherwise it will lead to a split-brain if only a single Patroni node in a cluster has this parameter set.

Actually, I don't see how the current option won't lead to a split-brain in a case like this (harder to reproduce locally on MacOS due to the need to isolate Etcd, should be rather easy on Linux with iptables):
- 2 patroni nodes are running against Etcd, with `static_single_node: true`
- the acting primary is isolated from Etcd (i.e. via the firewall), the replica still runs
- the replica waits until the leader key expires and promotes
- the acting primary doesn't demote, we get a split-brain.
-

Reply:
So, the main idea with this config value is actually that the user/operator of Patroni guarantees that the cluster will only ever have 1 node while this config value is set to `True`. So if this value is set to `True` and an additional node is added, then that is essentially UB.

In essence, this value has to be set to `False` in order to safely add new replicas.